crictl images fails with panic: runtime error: slice bounds out of range

With CRI-O build from its master (712f3b8cf14cbbac298f3ccc020677ac2a65fa75) and `cri-tools` built from its master (a6dd109ef238be53ab9a7a5c4642f8a1d4ba1bd6), running

```
# crictl pull registry.fedoraproject.org/fedora:27
```

passed but

```
# crictl images
```

fails with

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
main.glob..func13(0xc4200738c0, 0x0, 0xc4200738c0)
	/root/cri-tools/_output/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/image.go:155 +0x1002
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.HandleAction(0xca43a0, 0xe233f0, 0xc4200738c0, 0xc420055000, 0x0)
	/root/cri-tools/_output/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:490 +0xd2
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.Command.Run(0xdc4743, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0xdc9024, 0xb, 0x0, ...)
	/root/cri-tools/_output/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/command.go:210 +0xa95
github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli.(*App).Run(0xc420061a00, 0xc42000c060, 0x2, 0x2, 0x0, 0x0)
	/root/cri-tools/_output/src/github.com/kubernetes-incubator/cri-tools/vendor/github.com/urfave/cli/app.go:255 +0x6f8
main.main()
	/root/cri-tools/_output/src/github.com/kubernetes-incubator/cri-tools/cmd/crictl/main.go:218 +0xc71
```

Running

```
# crictl images --no-trunc
```

shows

```
IMAGE                               TAG                 IMAGE ID                                                           SIZE
<none>                              <none>                                                                                 0B
registry.fedoraproject.org/fedora   27                  99b71991af6eef73e85e3a657641cf2447929f37fff1f9570d525a6ef485a4a8   246MB
```

--- a/cmd/crictl/container.go
+++ b/cmd/crictl/container.go
@@ -619,7 +619,10 @@ func ListContainers(client pb.RuntimeServiceClient, opts listOptions) error {
 		if !opts.verbose {
 			id := c.Id
 			if !opts.noTrunc {
-				id = strings.TrimPrefix(c.Id, "")[:truncatedIDLen]
+				id = strings.TrimPrefix(c.Id, "")
+				if len(id) > truncatedIDLen {
+					id = id[:truncatedIDLen]
+				}
 			}
 			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%d\n",
 				id, c.Image.Image, ctm, c.State, c.Metadata.Name, c.Metadata.Attempt)

--- a/cmd/crictl/image.go
+++ b/cmd/crictl/image.go
@@ -152,7 +152,10 @@ var listImageCommand = cli.Command{
 				size := units.HumanSizeWithPrecision(float64(image.GetSize_()), 3)
 				id := image.Id
 				if !noTrunc {
-					id = strings.TrimPrefix(image.Id, "sha256:")[:truncatedIDLen]
+					id = strings.TrimPrefix(image.Id, "sha256:")
+					if len(id) > truncatedIDLen {
+						id = id[:truncatedIDLen]
+					}
 				}
 				for _, repoTagPair := range repoTagPairs {
 					if showDigest {

--- a/cmd/crictl/sandbox.go
+++ b/cmd/crictl/sandbox.go
@@ -429,7 +429,10 @@ func ListPodSandboxes(client pb.RuntimeServiceClient, opts listOptions) error {
 			ctm := units.HumanDuration(time.Now().UTC().Sub(createdAt)) + " ago"
 			id := pod.Id
 			if !opts.noTrunc {
-				id = strings.TrimPrefix(pod.Id, "")[:truncatedIDLen]
+				id = strings.TrimPrefix(pod.Id, "")
+				if len(id) > truncatedIDLen {
+					id = id[:truncatedIDLen]
+				}
 			}
 			fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%d\n",
 				id, ctm, pod.State, pod.Metadata.Name, pod.Metadata.Namespace, pod.Metadata.Attempt)

--- a/cmd/crictl/stats.go
+++ b/cmd/crictl/stats.go
@@ -161,7 +161,10 @@ func ContainerStats(client pb.RuntimeServiceClient, opts statsOptions) error {
 	// Use `+` to work around go vet bug
 	fmt.Fprintln(w, "CONTAINER\tCPU %"+"\tMEM\tDISK\tINODES")
 	for _, s := range r.GetStats() {
-		id := strings.TrimPrefix(s.Attributes.Id, "")[:truncatedIDLen]
+		id := strings.TrimPrefix(s.Attributes.Id, "")
+		if len(id) > truncatedIDLen {
+			id = id[:truncatedIDLen]
+		}
 		cpu := s.GetCpu().GetUsageCoreNanoSeconds().GetValue()
 		mem := s.GetMemory().GetWorkingSetBytes().GetValue()
 		disk := s.GetWritableLayer().GetUsedBytes().GetValue()

